api/python/cell_census/src/cell_census/open.py

```diff
@@ -17,11 +17,11 @@
 
 def _open_soma(locator: CensusLocator, context: Optional[soma.options.SOMATileDBContext] = None) -> soma.Collection:
     """Private. Merge config defaults and return open census as a soma Collection/context."""
-    context = context or soma.options.SOMATileDBContext()
     s3_region = locator.get("s3_region")
 
     if not context:
         # if no user-defined context, cell_census defaults take precedence over SOMA defaults
+        context = soma.options.SOMATileDBContext()
```

Comment:
I think we want this instead (untested):
```
        tiledb_config = {**DEFAULT_TILEDB_CONFIGURATION}
        if s3_region is not None:
            tiledb_config["vfs.s3.region"] = s3_region
        context = soma.options.SOMATileDBContext(tiledb_ctx=tiledb.Ctx(tilecb_config))
```

SOMATileDBContext() is supposed to be frozen, so I wouldn't expect the `replace()` to work.

Also `SOMATileDBContext` takes a tiledb.Ctx object, not a plain dict.

Whatever we end up with, I would verify the default tiledb config args are showing when

Reply:
I added a check for the tileDB defaults and they were set. The code also didn't complain about using replace.

Reply:
Thanks for confirming. I would still consider the revised code above since it's a smidge clearer.

Reply:
`tiledb` is an indirect dependency for the cell_census API. Adding `tiledb.Ctx` would require me to add it as a package requirement. Is that something we want?

https://github.com/chanzuckerberg/cell-census/blob/4485b64579b141b1b94e9005cb9f5fb5e26f638e/api/python/cell_census/pyproject.toml#L30-L42

Reply:
I don't see any other instance in the API where we are importing from the tiledb library directly